File: library/src/main/java/com/fanjun/keeplive/utils/ServiceUtils.java
package com.fanjun.keeplive.utils;

import android.app.ActivityManager;
import android.content.Context;

import java.util.Iterator;
import java.util.List;

public class ServiceUtils {
    public static boolean isServiceRunning(Context ctx, String className) {
        boolean isRunning = false;
        ActivityManager activityManager = (ActivityManager) ctx
                .getSystemService(Context.ACTIVITY_SERVICE);
        List<ActivityManager.RunningServiceInfo> servicesList = activityManager
                .getRunningServices(Integer.MAX_VALUE);
        if (servicesList != null) {
            Iterator<ActivityManager.RunningServiceInfo> l = servicesList.iterator();
            while (l.hasNext()) {
                ActivityManager.RunningServiceInfo si = l.next();
                if (className.equals(si.service.getClassName())) {
                    isRunning = true;
                }
            }
        }
        return isRunning;
    }
    public static boolean isRunningTaskExist(Context context, String processName) {
        ActivityManager am = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        List<ActivityManager.RunningAppProcessInfo> processList = am.getRunningAppProcesses();
        if (processList != null){
            for (ActivityManager.RunningAppProcessInfo info : processList) {
                if (info.processName.equals(processName)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: library/src/main/java/com/fanjun/keeplive/utils/ServiceUtils.kt
package com.fanjun.keeplive.utils

import android.app.ActivityManager
import android.content.Context

/**
 * 服务状态检测工具类
 * - isServiceRunning: 检查指定 Service 是否正在运行
 * - isRunningTaskExist: 检查指定进程是否存在
 */
@Suppress("DEPRECATION")
object ServiceUtils {

    /**
     * 检查指定类名的 Service 是否正在运行
     */
    @JvmStatic
    fun isServiceRunning(ctx: Context, className: String): Boolean {
        val activityManager = ctx.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val servicesList = activityManager.getRunningServices(Int.MAX_VALUE) ?: return false
        return servicesList.any { it.service.className == className }
    }

    /**
     * 检查指定进程名是否存在
     */
    @JvmStatic
    fun isRunningTaskExist(context: Context, processName: String): Boolean {
        val am = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val processList = am.runningAppProcesses ?: return false
        return processList.any { it.processName == processName }
    }
}
